src/mono/mono/metadata/native-library.c

```diff
@@ -1262,6 +1265,17 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 	addr = pinvoke_probe_for_symbol (module, piinfo, new_import, &error_msg);
 
 	if (!addr) {
+#if defined(ENABLE_NETCORE) && !defined(HOST_WIN32)
+		if (strcmp (new_scope, "__Internal") == 0) {
+			g_free ((char *)new_scope);
+#if defined(TARGET_OSX)
+			new_scope = g_strdup ("libcoreclr.dylib");
+#else			
+			new_scope = g_strdup ("libcoreclr.so");
```

Comment:
we should have a tracking item somewhere about revisiting this since we won't be using the libmono->libcoreclr rename hack forever.